sdk/samples/iot/paho_iot_hub_pnp_component_sample.c

```diff
@@ -50,12 +49,13 @@ static az_span const model_id
 // Components
 static pnp_thermostat_component thermostat_1;
 static pnp_thermostat_component thermostat_2;
-static az_span const thermostat_1_name = AZ_SPAN_LITERAL_FROM_STR("thermostat1");
-static az_span const thermostat_2_name = AZ_SPAN_LITERAL_FROM_STR("thermostat2");
-static az_span const device_information_name = AZ_SPAN_LITERAL_FROM_STR("deviceInformation");
-static az_span const* pnp_components[]
-    = { &thermostat_1_name, &thermostat_2_name, &device_information_name };
-static int32_t const pnp_components_num = sizeof(pnp_components) / sizeof(pnp_components[0]);
+static az_span thermostat_1_name = AZ_SPAN_LITERAL_FROM_STR("thermostat1");
+static az_span thermostat_2_name = AZ_SPAN_LITERAL_FROM_STR("thermostat2");
+static az_span pnp_device_components[] = { AZ_SPAN_LITERAL_FROM_STR("thermostat1"),
+                                           AZ_SPAN_LITERAL_FROM_STR("thermostat2"),
```

Comment:
Can you not use thermostat_1_name and thermostat_2_name here instead of recreating the az_span using AZ_SPAN_LITERAL_FROM_STR again?

Reply:
No unfortunately not since putting a variable in the literal is not constant.

Reply:
What about making the az_spans const?  It just looks funny having the repetition is all.

Reply:
Unfortunately still doesn't work
>**C99 6.7.8**
> 4. All the expressions in an initializer for an object that has static storage duration shall be
constant expressions or string literals

Only other thing I can think of is setting `AZ_SPAN_LITERAL_FROM_STR("thermostat1")` to a macro.

Reply:
gotcha. and I see now why it worked before, because it was an array of pointers, not az_spans.